Automattic/woocommerce-payments
# Add Bancontact as an UPE payment method

Fixes #2601 

#### Changes proposed in this Pull Request

* Add Bancontact method to UPE methods.
* Update tests accordingly.
* Refactor some tests to make them easier to manage for future methods.

#### Testing instructions

* Under dev tools the boxes for _Enable grouped settings_, _Enable UPE checkout_, and _Add UPE additional payment methods_ need to be checked.
* If EUR is your default currency, navigate to WooCommerce > Settings > General and change the currency to USD.
* If EUR is an enabled currency in your store, navigate to WooCommerce > Settings > Multi-Currency and remove EUR as a currency.
* Navigate to Payments > Settings.
* Click _Add payment method_, verify Bancontact is in the modal, along with its icon.
* Select Bancontact, a notice should appear stating that Euros are being added to your store. Click _Add selected_, then scroll to the bottom and click _Save changes_.
* Verify Bancontact and its logo are now in the _Payment methods_ area.
* Navigate to WooCommerce > Settings > Multi-Currency and verify EUR is now an enabled currency in your store.
* On the frontend, add an item to your cart, and go to checkout.
* Verify your currency *is not* EUR, payment method should show as _Credit card / debit card_ and when selecting _Use a new payment method_, Bancontact should not appear.
* Switch your currency to EUR, payment method should show as _Popular payment methods_ and when selecting _Use a new payment method_, Bancontact should appear.
* Select Bancontact.
* Complete checkout/place order.
* You will be taken to a Stripe test confirmation page, choose _Authorize test payment_.
* Once order is received, navigate to WooCommerce > Orders in the admin.
* Find your order and select it.
* Click _Refund_, and then increase a line item in your order to 1.
* Click _Refund via WooCommerce Payments_, verify your refund completed.
* Navigate to Payments > Transactions, you should see your _Payment_ and _Payment refund_ listed for your order, verify both have the Bancontact logo.
* Click on the Payment, then verify the _Payment method_ both at the top and bottom are correctly formatting (they will have default test info).
* Do the same for the Payment refund.
* Navigate to Payments > Settings, then click the trash can icon to delete Bancontact. Verify the name and the logo in the modal are correct, then click _Remove_, scroll to the bottom and _Save changes_.
* Verify Bancontact is no longer in the list of _Payment methods_.

#### Screenshots

![Markup 2021-08-13 at 15 13 27](https://user-images.githubusercontent.com/4634416/129402497-60086233-5a89-4fa2-942e-f93e56316460.png)
![Markup 2021-08-13 at 15 14 22](https://user-images.githubusercontent.com/4634416/129402512-6acf24a0-c8d4-41bd-b455-409f2b710321.png)
![Markup 2021-08-13 at 15 14 51](https://user-images.githubusercontent.com/4634416/129402524-cf31d1dd-623b-4d1a-84c9-3569658cfdda.png)
![Markup 2021-08-13 at 15 16 39](https://user-images.githubusercontent.com/4634416/129402536-64fd08cf-77ff-473a-b70b-48d809093ab1.png)
![Markup 2021-08-13 at 15 17 02](https://user-images.githubusercontent.com/4634416/129402550-4de4d342-8f1a-40e5-8732-88f38f11b678.png)
![Markup 2021-08-13 at 15 18 00](https://user-images.githubusercontent.com/4634416/129402557-8351de5a-cbe2-4a53-b494-b179daba35c9.png)
![Markup 2021-08-13 at 15 18 38](https://user-images.githubusercontent.com/4634416/129402570-aeef198b-a16b-4675-9a7c-d1d277f12564.png)


-------------------
